libraries/goclient/goLibrary.go

```diff
@@ -0,0 +1,126 @@
+package goclient
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/intel/afxdp-plugins-for-kubernetes/constants"
+
+	"github.com/intel/afxdp-plugins-for-kubernetes/internal/host"
+	"github.com/intel/afxdp-plugins-for-kubernetes/internal/uds"
+)
+
+var (
+	hWR          uds.Handler
+	hPod         host.Handler
+	busy_timeout int = 20
+	busy_budget  int = 64
+)
+
+const (
+	requestDelay = 100 * time.Millisecond
+)
+
+/*
+Returns the version of our Handshake as a string
+*/
+func GetVersionStr() string {
+	return constants.Uds.Handshake.Version
+}
+
+/*
+Call this function to initialize the library, returns a cleanup function
+*/
+func Init() uds.CleanupFunc {
+	hWR = uds.NewHandler()
+	hPod = host.NewHandler()
+
+	// init uds Handler for reading and writing
+	if err := hWR.Init(constants.Uds.PodPath, constants.Uds.Protocol, constants.Uds.MsgBufSize, constants.Uds.CtlBufSize, 0*time.Second, ""); err != nil {
+		println("Library Error: Error Initialising UDS server: ", err)
+		os.Exit(1)
+	}
+
+	cleanup, err := hWR.Dial()
+	if err != nil {
+		println("Library Error: UDS Dial error:: ", err)
+		cleanup()
+		os.Exit(1)
+	}
+	hostname, err := hPod.Hostname()
+	if err != nil {
+		println("Library Error: Failed to initialize host: ", err)
+		os.Exit(1)
+	}
+	makeRequest("/connect, "+hostname, -1)
+	return cleanup
+}
+
+/*
+Returns the version of our Handshake
+*/
+func GetVersion() {
+	makeRequest("/version", -1)
+}
+
+/*
+Give it a list of device names and returns a map of the fds for each device and a cleanup function to close the connection
+*/
+func RequestXSKmapFD(devNames []string) map[string]int {
+	fds := make(map[string]int)
+
+	for _, dev := range devNames {
+		fd := makeRequest("/xsk_map_fd, "+dev, -1)
```

Comment:
The string requests that make up the handshake are all defined as constants here:
https://github.com/intel/afxdp-plugins-for-kubernetes/blob/main/constants/constants.go#L320
and the strings are actually here:
https://github.com/intel/afxdp-plugins-for-kubernetes/blob/main/constants/constants.go#L82

If you move to using these constants `constants.Uds.Handshake.X` it means if we ever update the protocol then we won't have to come back here an update these hardcoded strings within your library.